environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java

```diff
@@ -170,6 +197,31 @@ private EnvironmentDto changeCredential(String accountId, String environmentName
         return environmentDtoConverter.environmentToDto(saved);
     }
 
+    private EnvironmentDto updateAwsDiskEncryptionParameters(String accountId, String environmentName, AwsDiskEncryptionParametersDto dto,
+            Environment environment) {
+            ValidationResult validateKey = environmentService.getValidatorService().validateEncryptionKeyArn(dto.getEncryptionKeyArn(),
+                    accountId);
```

Comment:
I think `dto.getEncryptionKeyArn()` can be `null`, so perhaps the method logic should be rearranged and omit calling `EnvironmentValidatorService.validateEncryptionKeyArn()` in that scenario.